Error message of `TERM=dumb zig run x.zig` is missing more than just color

### Zig Version

0.10.0-dev.2836+2360f8c49

### Steps to Reproduce

```zig
pub fn main() {}
```
```
TERM=dumb zig run x.zig
```

### Expected Behavior

```
x.zig:1:15: error: expected return type expression, found '{'
pub fn main() {}
              ^
```
(without color)

### Actual Behavior

```
x.zig:1:15: error: expected return type expression, found '{'
```
(no color (expected) and the error message is missing the source code and the caret (unexpected))

The output for dumb terminals can be improved to be on par with the output for non-dumb terminals, without escape sequences.

GCC gets it right:
![image](https://user-images.githubusercontent.com/35064754/178539846-c19b7192-f98f-4684-bc01-1b09d70658a1.png)

--- a/src/Compilation.zig
+++ b/src/Compilation.zig
@@ -429,25 +429,23 @@ pub const AllErrors = struct {
                         try stderr.print(" ({d} times)\n", .{src.count});
                     }
                     ttyconf.setColor(stderr, .Reset);
-                    if (ttyconf != .no_color) {
-                        if (src.source_line) |line| {
-                            for (line) |b| switch (b) {
-                                '\t' => try stderr.writeByte(' '),
-                                else => try stderr.writeByte(b),
-                            };
-                            try stderr.writeByte('\n');
-                            // TODO basic unicode code point monospace width
-                            const before_caret = src.span.main - src.span.start;
-                            // -1 since span.main includes the caret
-                            const after_caret = src.span.end - src.span.main -| 1;
-                            try stderr.writeByteNTimes(' ', src.column - before_caret);
-                            ttyconf.setColor(stderr, .Green);
-                            try stderr.writeByteNTimes('~', before_caret);
-                            try stderr.writeByte('^');
-                            try stderr.writeByteNTimes('~', after_caret);
-                            try stderr.writeByte('\n');
-                            ttyconf.setColor(stderr, .Reset);
-                        }
+                    if (src.source_line) |line| {
+                        for (line) |b| switch (b) {
+                            '\t' => try stderr.writeByte(' '),
+                            else => try stderr.writeByte(b),
+                        };
+                        try stderr.writeByte('\n');
+                        // TODO basic unicode code point monospace width
+                        const before_caret = src.span.main - src.span.start;
+                        // -1 since span.main includes the caret
+                        const after_caret = src.span.end - src.span.main -| 1;
+                        try stderr.writeByteNTimes(' ', src.column - before_caret);
+                        ttyconf.setColor(stderr, .Green);
+                        try stderr.writeByteNTimes('~', before_caret);
+                        try stderr.writeByte('^');
+                        try stderr.writeByteNTimes('~', after_caret);
+                        try stderr.writeByte('\n');
+                        ttyconf.setColor(stderr, .Reset);
                     }
                     for (src.notes) |note| {
                         try note.renderToWriter(ttyconf, stderr, "note", .Cyan, indent);

--- a/src/test.zig
+++ b/src/test.zig
@@ -1662,6 +1662,7 @@ pub const TestContext = struct {
                                         var msg: Compilation.AllErrors.Message = actual_error;
                                         msg.src.src_path = case_msg.src.src_path;
                                         msg.src.notes = &.{};
+                                        msg.src.source_line = null;
                                         var fib = std.io.fixedBufferStream(&buf);
                                         try msg.renderToWriter(.no_color, fib.writer(), "error", .Red, 0);
                                         var it = std.mem.split(u8, fib.getWritten(), "error: ");

